### PR TITLE
Theme inheritance

### LINKED ISSUE #173: Theme inheritance restricted to same source tree
Building a theme which inherits from an existing implementation (not directly from wcomponents-theme) currently requires that the sub-theme and inherited theme are in the same source directory, for example:

```
src
 |- parent_theme
 |_ child_theme
```

This is overly restrictive as it prevents use of published themes (such as our own theme-inhouse project) as the basis of inheritance. I would argue that this is a bug rather than an enhancement because we have a sample theme which allows inheritance (through the existence of `sources.xml`) but which is not usable for this very process.

## Suggestion

We should allow inheritance from a path to another theme, including from the source zip file created during build. So given two independent paths:
```
/some/path/to/parent_theme
/some/other/path/to/child_theme
```

The child theme should be buildable having an `inherit.txt` containing either `/some/path/to/parent_theme` **or** `/some/path/to/a/previously/built/parent/theme/parent_theme-[VERSION]-sources.zip`.

### PR DESCRIPTION
**Pull request by marksreeves**
Updated sub-theme merging so that `inherit.txt` can contain one of:

1. a simple string without “/“ which is a theme in the current source
root (the currently supported scenario); or
2. the path to the root directory of the theme to be inherited; or
3. the path to a theme sources zip file.

Fixes #173

### CODE REVIEW

**Comment by marksreeves** [comment]
@ricksbrown Do you want to take a look at this before it is merged?

File: wcomponents-theme/build.xml [hunk 1]
```diff
@@ -233,15 +233,59 @@ chromedriver_cdnurl=${mirror.root}/chromedriver
...
+							<then>
+								<echo>Found prototype directory named ${prototype.name}</echo>
+								<property name="prototype.dir" location="${prototype.name}"/>
+							</then>
+						</elseif>
+						<elseif>
+							<available file="${impl.src.rootdir}/${prototype.name}" type="dir"/>
+							<then>
+								<echo>Using simple fallback prototype ${impl.src.rootdir}/${prototype.name}</echo>
+								<property name="prototype.dir" location="${impl.src.rootdir}/${prototype.name}"/>
```
Comment (ricksbrown) [inline wcomponents-theme/build.xml]: Same deal here, may as well set `prototype.dir` at the top of this block

File: wcomponents-theme/build.xml [hunk 2]
```diff
@@ -233,15 +233,59 @@ chromedriver_cdnurl=${mirror.root}/chromedriver
...
+						<then>
+							<echo>Found prototype zip file ${prototype.name}</echo>
+							<delete dir="${tmp.dir}/parent"/>
+							<!-- unzip the zip and set the path -->
+							<unzip src="${prototype.name}" dest="${tmp.dir}/parent" overwrite="true" failOnEmptyArchive="true">
+								<patternset>
+									<include name="*/**"/>
+								</patternset>
+							</unzip>
+							<property name="prototype.dir" location="${tmp.dir}/parent"/>
```
Comment (ricksbrown) [inline wcomponents-theme/build.xml]: Suggest moving this to the top of the `then` block and then replace usages of `${tmp.dir}/parent` with `${prototype.dir}`.

**Comment by ricksbrown** [comment]
@marksreeves two optional suggestions, otherwise LGTM
In future we may want to consider allowing relative paths as well but for now it's not worth it.

**Pull request merged**